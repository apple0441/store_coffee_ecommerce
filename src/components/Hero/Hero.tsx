import { Box,
    Center,    
    Heading, 
    Link, 
    List, 
    ListItem, 
    SimpleGrid,
    Text,
    useColorModeValue } from "@chakra-ui/react";
import NextImage from "next/image";
import NextLink from "next/link";
import * as React from "react";

const Hero = (): React.ReactElement => {
 return (
    <Box 
    as="header" 
    position="relative"
    >
        <Center 
        as="article"
        display="flex"
        flexDirection="column"
        h="55vh"
        >
        <SimpleGrid 
        bottom="0"
        columns={3}
        left="0"
        position="absolute"
        right="0"
        spacing={8}
        top="0"
        zIndex="-10"
        >
            <Box bg={useColorModeValue("white", "gray.800")} />
            <Box bg={useColorModeValue("white", "gray.800")} />
            <Box bg={useColorModeValue("white", "gray.800")} />
            <Box 
            bg={useColorModeValue("pink.50", "gray.700")}
            position="relative"
            opacity="0.3"
            >
                <NextImage
                alt=""
                src="/coffee-product-8.webp"
                layout="fill"
                objectFit="cover"
                objectPosition="bottom"
                />
            </Box>
            <Box bg={useColorModeValue("white", "gray.800")} />
            <Box 
            bg={useColorModeValue("pink.50", "gray.700")} 
            position="relative"
            opacity="0.3"
            >
                <NextImage
                alt=""
                src="/coffee-product-4.webp"
                layout="fill"
                objectFit="cover"
                />
            </Box>
        </SimpleGrid>
        <Heading 
        as="h1"
        fontWeight="400"
        mx="auto"
        mb={8}
        textAlign="center"
        textTransform="uppercase"
        w="90vw"
        >
        Bringing you the best coffee from all over the world
        </Heading>
            <List 
            alignItems="center"
            display={["none", "none", "flex"]}
            flexDirection="row"
            justifyContent="space-evenly"
            textAlign="center"
            my={8}
            w="75vw"
            >
                <ListItem
                bg={useColorModeValue("white", "gray.600")}
                fontWeight="700"
                py={1}
                px={2}
                textTransform="uppercase"
                w="10rem"
                >
                    Coffee Beans
                </ListItem>
                <ListItem
                bg={useColorModeValue("white", "gray.600")}
                fontWeight="700"
                py={1}
                px={2}
                textTransform="uppercase"
                w="10rem"
                >
                    Ground Coffee
                </ListItem>
                <ListItem
                bg={useColorModeValue("white", "gray.600")}
                fontWeight="700"
                py={1}
                px={2}
                textTransform="uppercase"
                w="10rem"
                >
                    Mugs & Cups
                </ListItem>
            </List>
            <NextLink 
            href="/store"
            passHref
            >
                <Link
                bg="pink.400"
                color="white"
                borderRadius="md"
                fontWeight="700"
                py={3}
                px={6}
                textAlign="center"
                textTransform="uppercase"
                w="12rem">
                    Shop All Items
                </Link>
            </NextLink>
        </Center>
    </Box>
 );
};

export default Hero;