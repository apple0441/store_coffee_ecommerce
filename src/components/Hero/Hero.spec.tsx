import { render, RenderResult } from "@testing-library/react";
import * as React from "react";
import TestRenderer from "react-test-renderer";

import Hero from "@/components/Hero/Hero";

let documentBody: RenderResult;

describe("<Hero />", () => {
    test("renders", () => {
        documentBody = render(<Hero />);

        const h1Element = documentBody.getByRole("heading", { 
            name: "Bringing you the best coffee from all over the world"
        });

        expect(h1Element).toBeInTheDocument();
        expect(documentBody.getByRole("article")).toContainElement(h1Element);
    });

    test("snapshot", () => {
        const tree = TestRenderer
        .create(<Hero />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});