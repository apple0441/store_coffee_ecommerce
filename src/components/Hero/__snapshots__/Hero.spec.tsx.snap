// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> snapshot 1`] = `
<div
  className="css-0"
>
  <article
    className="css-1fksumk"
  >
    <h1
      className="chakra-heading css-1ghmqg6"
    >
      Bringing you the best coffee from all over the world
    </h1>
    <ul
      className="css-1imb5xo"
      role="list"
    >
      <li
        className="css-1m90wc1"
      >
        <a
          className="chakra-link css-0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Coffee Beans
        </a>
      </li>
      <li
        className="css-1m90wc1"
      >
        <a
          className="chakra-link css-0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Ground Coffee
        </a>
      </li>
      <li
        className="css-1m90wc1"
      >
        <a
          className="chakra-link css-0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Mugs & Cups
        </a>
      </li>
    </ul>
    <a
      className="chakra-link css-yjd4r4"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Shop All Items
    </a>
  </article>
</div>
`;
